dali/python/backend_impl.cc

```diff
@@ -644,7 +644,7 @@ PYBIND11_MODULE(backend_impl, m) {
         [](Pipeline *p, const std::vector<std::pair<string, string>>& outputs) {
           p->SetOutputNames(outputs);
           })
-    .def("RunCPU", &Pipeline::RunCPU)
+    .def("RunCPU", &Pipeline::RunCPU, py::call_guard<py::gil_scoped_release>())
```

Comment:
I am just wondering what is this extra argument doing. Any performance implications (since this affects any pipeline's RunCPU)

Reply:
What it does is releasing the python global interpreter lock. Since a normal pipeline execution does not touch any python's resources it is safe to do it while not holding the GIL. Speaking about performance, actually it could even help because it unlocks the interpreter for other python threads. Of course in case of a python operator, GIL must be reacquired before  executing.